fast64_internal/oot/oot_constants.py

```diff
@@ -44,449 +55,29 @@
-	("ACTOR_DEMO_GEFF", "DEMO_GEFF", "DEMO_GEFF"),
-	("ACTOR_BG_GND_FIREMEIRO", "BG_GND_FIREMEIRO", "BG_GND_FIREMEIRO"),
-	("ACTOR_BG_GND_DARKMEIRO", "BG_GND_DARKMEIRO", "BG_GND_DARKMEIRO"),
-	("ACTOR_BG_GND_SOULMEIRO", "BG_GND_SOULMEIRO", "BG_GND_SOULMEIRO"),
-	("ACTOR_BG_GND_NISEKABE", "BG_GND_NISEKABE", "BG_GND_NISEKABE"),
-	("ACTOR_BG_GND_ICEBLOCK", "BG_GND_ICEBLOCK", "BG_GND_ICEBLOCK"),
-	("ACTOR_EN_GB", "EN_GB", "EN_GB"),
-	("ACTOR_EN_GS", "EN_GS", "EN_GS"),
-	("ACTOR_BG_MIZU_BWALL", "BG_MIZU_BWALL", "BG_MIZU_BWALL"),
-	("ACTOR_BG_MIZU_SHUTTER", "BG_MIZU_SHUTTER", "BG_MIZU_SHUTTER"),
-	("ACTOR_EN_DAIKU_KAKARIKO", "EN_DAIKU_KAKARIKO", "EN_DAIKU_KAKARIKO"),
-	("ACTOR_BG_BOWL_WALL", "BG_BOWL_WALL", "BG_BOWL_WALL"),
-	("ACTOR_EN_WALL_TUBO", "EN_WALL_TUBO", "EN_WALL_TUBO"),
-	("ACTOR_EN_PO_DESERT", "EN_PO_DESERT", "EN_PO_DESERT"),
-	("ACTOR_EN_CROW", "EN_CROW", "EN_CROW"),
-	("ACTOR_DOOR_KILLER", "DOOR_KILLER", "DOOR_KILLER"),
-	("ACTOR_BG_SPOT11_OASIS", "BG_SPOT11_OASIS", "BG_SPOT11_OASIS"),
-	("ACTOR_BG_SPOT18_FUTA", "BG_SPOT18_FUTA", "BG_SPOT18_FUTA"),
-	("ACTOR_BG_SPOT18_SHUTTER", "BG_SPOT18_SHUTTER", "BG_SPOT18_SHUTTER"),
-	("ACTOR_EN_MA3", "EN_MA3", "EN_MA3"),
-	("ACTOR_EN_COW", "EN_COW", "EN_COW"),
-	("ACTOR_BG_ICE_TURARA", "BG_ICE_TURARA", "BG_ICE_TURARA"),
-	("ACTOR_BG_ICE_SHUTTER", "BG_ICE_SHUTTER", "BG_ICE_SHUTTER"),
-	("ACTOR_EN_KAKASI2", "EN_KAKASI2", "EN_KAKASI2"),
-	("ACTOR_EN_KAKASI3", "EN_KAKASI3", "EN_KAKASI3"),
-	("ACTOR_OCEFF_WIPE4", "OCEFF_WIPE4", "OCEFF_WIPE4"),
-	("ACTOR_EN_EG", "EN_EG", "EN_EG"),
-	("ACTOR_BG_MENKURI_NISEKABE", "BG_MENKURI_NISEKABE", "BG_MENKURI_NISEKABE"),
-	("ACTOR_EN_ZO", "EN_ZO", "EN_ZO"),
-	("ACTOR_OBJ_MAKEKINSUTA", "OBJ_MAKEKINSUTA", "OBJ_MAKEKINSUTA"),
-	("ACTOR_EN_GE3", "EN_GE3", "EN_GE3"),
-	("ACTOR_OBJ_TIMEBLOCK", "OBJ_TIMEBLOCK", "OBJ_TIMEBLOCK"),
-	("ACTOR_OBJ_HAMISHI", "OBJ_HAMISHI", "OBJ_HAMISHI"),
-	("ACTOR_EN_ZL4", "EN_ZL4", "EN_ZL4"),
-	("ACTOR_EN_MM2", "EN_MM2", "EN_MM2"),
-	("ACTOR_BG_JYA_BLOCK", "BG_JYA_BLOCK", "BG_JYA_BLOCK"),
-	("ACTOR_OBJ_WARP2BLOCK", "OBJ_WARP2BLOCK", "OBJ_WARP2BLOCK"),
-]
+# Add every actor
+ootEnumActorID = [(actorNode.get('Key'), actorNode.get('Name'),
+					actorNode.get('ID').replace('ACTOR_','')) for actorNode in root if actorNode.get('ID') is not None]
+ootEnumActorID.insert(0, ("Custom", "Custom Actor", "Custom"))
+
+# Old actor list, we can't delete this (for now) as it'd create compatibility issues
+ootEnumActorIDLegacy = [("ACTOR_" + (actorNode.get('Key').upper()), actorNode.get('Name'),
+					actorNode.get('ID').replace('ACTOR_','')) for actorNode in root if actorNode.get('ID') is not None]
+ootEnumActorIDLegacy.insert(0, ("Custom", "Custom Actor", "Custom"))
+
+ootChestContent = [(elem.get('Value'), elem.get('Name'),
+					elem.get('Name')) for listNode in root for elem in listNode if listNode.tag == 'List'
+					and listNode.get('Name') == 'Chest Content']
+
+ootNaviMsgID = [(elem.get('Value'), elem.get('Name'),
```

Comment:
the identifier in hex works but as for actors it isn't great if a mod ever shifts the values

------------

also why is `itemDrops`, same as this but with the `Collectibles` `<List>`, in `editOOTActorProperties` instead of here (they probably should all be together but idk)

actually why is Collectibles at all different from ChestContent

Reply:
"collectibles" is whatever ``en_item00`` can give you, "ChestContent" is whatever items a chest can have, both are not the same

about ``itemDrops`` I guess I put it in ``editOOTActorProperties`` because it's only used there

Reply:
why is itemDrops only used in `editOOTActorProperties` with `genEnum`, while ChestContent is used by `OOT_SearchChestContentEnumOperator`?